lib/generators/solidus_pay_tomorrow/install/templates/solidus_pay_tomorrow_order_application_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module SolidusPayTomorrow
+  class OrderApplicationController < Spree::StoreController
+    def success
+      # FIXME: will have to find the correct payment object properly.
+      # This is error prone
+      payment = current_order.payments.where(state: :checkout).last
```

Comment:
As we already discussed, we have to link this payment with the PaymentSource when the payment is created, so we can filter the payments by the `source`.

Reply:
Added PaymentSource in [this PR](https://github.com/nebulab/solidus_pay_tomorrow/pull/12) and fixed the query here in [this commit](https://github.com/nebulab/solidus_pay_tomorrow/pull/11/commits/eff5820dc37118df52146515590a7da2392260cc)